### PR TITLE
feat: Initial molgx integration

### PR DESCRIPTION
**Pull request by kishimoto1974**
A first integration of MolGX (currently provided as a separate package) with GT4SD

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -356,6 +356,8 @@ Beyond implementing various generative modeling inference and training pipelines
 - [diffusers](https://github.com/huggingface/diffusers): training and inference pipelines for generative models from [Diffusers Models](https://github.com/huggingface/diffusers)
 - [GFlowNets](https://github.com/recursionpharma/gflownet): training and inference pipeline for [Generative Flow Networks](https://yoshuabengio.org/2022/03/05/generative-flow-networks/)
 
+- [MolGX](https://github.com/GT4SD//molgx/): training and inference pipelines to generate small molecules satisfying target properties. The full functions of MolGX are available [here](https://github.com/GT4SD/molgx/).
```
Comment (drugilsberg) [inline README.md:359]:
```suggestion
- [MolGX](https://github.com/GT4SD//molgx/): training and inference pipelines to generate small molecules satisfying target properties. The full implementation of MolGX, including additional functionalities, is available [here](https://github.com/GT4SD/molgx/). 
```

File: src/gt4sd/algorithms/conditional_generation/molgx/core.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+    def sample(self, number_of_items: int = 100) -> Iterator[S]:
+        """Generate a number of unique and valid items.
+
+        Args:
+        number_of_items: number of items to generate.
+                Defaults to 100.
+
+        Yields:
+                the items.
```
Comment (drugilsberg) [inline src/gt4sd/algorithms/conditional_generation/molgx/core.py:113]:
```suggestion
            number_of_items: number of items to generate. Defaults to 100.

        Yields:
            the items.
```

**Review by drugilsberg — COMMENTED** [review]
Only minor suggestions on the docstrings/README.md.

File: README.md (hunk 1 above)
Comment (kishimoto1974) [reply README.md]: I have also noticed an incorrect link to MolGX so have fixed that issue at the same time, while reflecting the above change.

**Review by drugilsberg — APPROVED** [review]
Good to go 🚀

**Pull request merged**